lucene/core/src/java/org/apache/lucene/util/fst/Builder.java

```diff
@@ -205,22 +156,145 @@ public Builder(FST.INPUT_TYPE inputType, int minSuffixCount1, int minSuffixCount
   }
 
   /**
-   * Overrides the default the maximum oversizing of fixed array allowed to enable direct addressing
-   * of arcs instead of binary search.
+   * Fluent-style constructor for FST {@link Builder}.
    * <p>
-   * Setting this factor to a negative value (e.g. -1) effectively disables direct addressing,
-   * only binary search nodes will be created.
-   *
-   * @see #DIRECT_ADDRESSING_MAX_OVERSIZING_FACTOR
+   * Creates an FST/FSA builder with all the possible tuning and construction tweaks.
+   * Read parameter documentation carefully.
    */
-  public Builder<T> setDirectAddressingMaxOversizingFactor(float factor) {
-    directAddressingMaxOversizingFactor = factor;
-    return this;
+  public static class Constructor<T> {
```

Comment:
re name to builder

Reply:
I agree - Constructor is easily confused with java.util.Constructor

Reply:
Oh o.a.l.util.fst.FSTCompiler or o.a.l.util.fst.Compiler - I guess I have a slight preference for the latter, but up to you

Reply:
I have a preference for fst.FSTCompiler. This may seem redundant when looking at the full path, but inside some code it is clearer to see FSTCompiler than just Compiler. So I'll go with FSTCompiler.